src/App.tsx

```diff
@@ -1,61 +1,122 @@
 import './App.scss';
+import { FC, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { prepareTodos, getNewTodoId, getUserById } from './helpers';
+import { TodoWithUser } from './types/types';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const preparedTodos = prepareTodos(todosFromServer, usersFromServer);
+
+export const App: FC = () => {
+  const [userSelect, setUserSelect] = useState(0);
+  const [todos, setTodos] = useState(preparedTodos);
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [usersError, setUsersError] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (!userSelect) {
+      setUsersError(true);
+    }
+
+    if (!userSelect || !title) {
+      return;
+    }
+
+    const newTodo: TodoWithUser = {
+      id: getNewTodoId(todos),
+      completed: false,
+      user: getUserById(usersFromServer, Number(userSelect)),
+      title: title.trim(),
+      userId: Number(userSelect),
+    };
+
+    setTodos([...todos, newTodo]);
```

Comment:
good practice to use a callback when we need a `prevState`
```suggestion
    setTodos((prevTodos) => [...prevTodos, newTodo]);
```